divide_and_conquer/heaps_algorithm_iterative.py

```diff
@@ -0,0 +1,57 @@
+"""
+Heap's algorithm (iterative) returns the list of all permutations possible from a list.
+It minimizes movement by generating each permutation from the previous one
+by swapping only two elements.
+More information:
+https://en.wikipedia.org/wiki/Heap%27s_algorithm.
+"""
+
+
+def heaps(arr: list) -> list:
+    """
+    Pure python implementation of the iterative Heap's algorithm,
+    returning all permutations of a list.
+    >>> heaps([])
+    [()]
+    >>> heaps([0])
+    [(0,)]
+    >>> heaps([-1, 1])
+    [(-1, 1), (1, -1)]
+    >>> heaps([1, 2, 3])
+    [(1, 2, 3), (2, 1, 3), (3, 1, 2), (1, 3, 2), (2, 3, 1), (3, 2, 1)]
+    >>> from itertools import permutations
+    >>> sorted(heaps([1,2,3])) == sorted(permutations([1,2,3]))
+    True
+    """
```

Comment:
```suggestion
    >>> all(sorted(heaps(x)) == sorted(permutations(x)) for x in ([], [0], [-1, 1], [1, 2, 3]))
    True
    """
```

Reply:
Then I think we are ready to land this one.

Reply:
Modified the iterative version as well as the recursive one ([#2475 ](https://github.com/TheAlgorithms/Python/pull/2475)) with your suggestion. 
Thanks for reviewing ! 👍